### PR TITLE
Add lax implementation of np.indices

### PR DESCRIPTION
**Pull request by jenkspt**
This implements the numpy `indices` function in `jax/numpy/lax_numpy.py`.

Need additional documentation or help to figure out how to implement the test case.

### CODE REVIEW

**Comment by jakevdp** [comment]
Looks good! A couple comments:

- you need to add ``indices`` to the list of imports in https://github.com/google/jax/blob/master/jax/numpy/__init__.py#L20-L58
- for testing, you can copy one of the existing tests in https://github.com/google/jax/blob/master/tests/lax_numpy_test.py and modify it to suit this case. Since there are no input arrays, the test should be much simpler than most.

File: jax/numpy/lax_numpy.py [hunk 1]
```diff
@@ -2250,6 +2250,20 @@ def ix_(*args):
...
+@_wraps(onp.indices)
+def indices(dimensions, dtype=int32, sparse=False):
+  dimensions = tuple(dimensions)
+  N = len(dimensions)
+  output = []
+  s = dimensions
+  for i, dim in enumerate(dimensions):
+    idx = lax.iota(dtype, dim)
+    if sparse:
+      s = [1]*N
```
Comment (jakevdp) [inline jax/numpy/lax_numpy.py:2262]: Use ``ones(N, dtype=dtype)``

Comment (jakevdp) [reply jax/numpy/lax_numpy.py:2262]: on second thought, a tuple would be the right thing to use here.

Comment (jenkspt) [reply jax/numpy/lax_numpy.py]: done!

**Review by jakevdp — APPROVED** [review]
Looks good, thanks!

**Comment by gnecula** [comment]
I had to adjust the test to pass on version 1.16.4 (the internal version in Google). See #3053.

**Pull request merged**